### PR TITLE
Add coverage for create location functionality for API

### LINKED ISSUE #2426: Automate location create in api
please automate location create in api

POST /api/locations
Create a location
Params
Param name 	Description
location
required 	
Value: Hash
location[name]
required 	
Value: String
location[description]
optional , nil allowed 	
Value: String
location[user_ids]
optional , nil allowed 	

User IDs

Value: Must be an array of any type
location[smart_proxy_ids]
optional , nil allowed 	

Smart proxy IDs

Value: Must be an array of any type
location[compute_resource_ids]
optional , nil allowed 	

Compute resource IDs

Value: Must be an array of any type
location[media_ids]
optional , nil allowed 	

Media IDs

Value: Must be an array of any type
location[config_template_ids]
optional , nil allowed 	

Provisioning template IDs

Value: Must be an array of any type
location[domain_ids]
optional , nil allowed 	

Domain IDs

Value: Must be an array of any type
location[realm_ids]
optional , nil allowed 	

Realm IDs

Value: Must be an array of any type
location[hostgroup_ids]
optional , nil allowed 	

Host group IDs

Value: Must be an array of any type
location[environment_ids]
optional , nil allowed 	

Environment IDs

Value: Must be an array of any type
location[subnet_ids]
optional , nil allowed 	

Subnet IDs

Value: Must be an array of any type

### PR DESCRIPTION
**Pull request by oshtaier**
Implemented most of necessary cases for create location functionality

Closes #2426

```
nosetests tests/foreman/api/test_location.py
.................
----------------------------------------------------------------------
Ran 17 tests in 76.922s
```

### CODE REVIEW

File: tests/foreman/api/test_location.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+http://theforeman.org/api/apidoc/v2/locations.html
+
+"""
+from ddt import ddt
+from fauxfactory import gen_string
+from nailgun import entities
+from random import randint
+from robottelo.common.decorators import data
+from robottelo.test import APITestCase
+# (too-many-public-methods) pylint:disable=R0904
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: Current versions of Pylint know to ignore all the methods inherited from unittest, or something like that. As a result, you can just drop this line.

By the way, an even nicer way to write this is:

    # pylint:disable=too-many-public-methods

File: tests/foreman/api/test_location.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        @Assert: Location created successfully and has correct user assigned to
+        it with expected login name
+
+        @Feature: Location
+
+        """
+        user = entities.User().create()
+        location = entities.Location(user=[user.id]).create()
+        self.assertEqual(location.user[0].id, user.id)
+        self.assertEqual(location.read_json()['users'][0]['login'], user.login)
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: You can also do this:

```python
self.assertEqual(

File: tests/foreman/api/test_location.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        @Feature: Location
+
+        """
+        test_cr = entities.LibvirtComputeResource().create()
+        location = entities.Location(compute_resource=[test_cr.id]).create()
+        self.assertEqual(location.compute_resource[0].id, test_cr.id)
+        self.assertEqual(
+            location.read_json()['compute_resources'][0]['provider'],
+            'Libvirt'
+        )
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: You can also do this:

```python
self.assertEqual(location.compute_resource[0].read().provider, 'Libvirt')
```

File: tests/foreman/api/test_location.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        @Feature: Location
+
+        """
+        test_cr = entities.DockerComputeResource().create()
+        location = entities.Location(compute_resource=[test_cr.id]).create()
+        self.assertEqual(location.compute_resource[0].id, test_cr.id)
+        self.assertEqual(
+            location.read_json()['compute_resources'][0]['provider'],
+            'Docker'
+        )
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: Same comment as above.

File: tests/foreman/api/test_location.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        """@Test: Create new location with assigned user to it
+
+        @Assert: Location created successfully and has correct user assigned to
+        it with expected login name
+
+        @Feature: Location
+
+        """
+        user = entities.User().create()
+        location = entities.Location(user=[user.id]).create()
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: This works just fine, but it's even more idiomatic to write `user=[user]`

File: tests/foreman/api/test_location.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        it
+
+        @Assert: Location created successfully and has correct Libvirt compute
+        resource assigned to it
+
+        @Feature: Location
+
+        """
+        test_cr = entities.LibvirtComputeResource().create()
+        location = entities.Location(compute_resource=[test_cr.id]).create()
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: This works fine, but it's even more idiomatic to write `compute_resource=[test_cr]`.

File: tests/foreman/api/test_location.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        it
+
+        @Assert: Location created successfully and has correct Docker compute
+        resource assigned to it
+
+        @Feature: Location
+
+        """
+        test_cr = entities.DockerComputeResource().create()
+        location = entities.Location(compute_resource=[test_cr.id]).create()
```
Comment (Ichimonji10) [inline tests/foreman/api/test_location.py]: Same comment as above.

**Comment by Ichimonji10** [comment]
Looks great. All my comments can be considered nitpicks. Fix them if you have time, but go ahead and merge this if a second ACK is received and having this test is time-sensitive.

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/api/test_location.py
.................
----------------------------------------------------------------------
Ran 17 tests in 85.262s

OK
```

**Comment by Ichimonji10** [comment]
Looks fantastic. Thank you. ACK

**Comment by elyezer** [comment]
ACK. Thank you.

**Pull request merged**